File: WebContent/cqut/hrms/view/post/js/PostService.js
var postApp = angular.module('postApp', []);

postApp.factory('PostService',['$resource',function($resource){
	
	var postService=$resource('/Post/:action', {});
	
	postService.insertPost=function(post,sucesscb,errorcb){
		
		postService.save({action:'insertPost'},post,sucesscb,errorcb);
	};
	
	postService.getAllPosts = function(sucesscb,errorcb){
		
		postService.query({action:'getAllPosts'},sucesscb,errorcb);
	}
	
	postService.getPostById=function(postId,sucesscb,errorcb){
		
		postService.save({action:'getPostById'},postId,sucesscb,errorcb);
	}
	
	postService.updatePost = function(post,sucesscb,errorcb){
		
		postService.save({action:'updatePost'},post,sucesscb,errorcb);
		
	};
	
	postService.deletePost =function(postId,sucesscb,errorcb){
		
		postService.save({action:'deletePost'},postId,sucesscb,errorcb);
		
	};
	
	postService.getPostByDepartmentIdAndName=function(departmentId,name,sucesscb,errorcb){
		
		postService.save({action:'getPostByDepartmentIdAndName'},{departmentId:departmentId,name:name},sucesscb,errorcb);
		
	};
	
	postService.getPostByDepartmentId=function(departmentId,sucesscb,errorcb){
		
		postService.save({action:'getPostByDepartmentId'},departmentId,sucesscb,errorcb);
		
	};
	
	postService.getPostByName=function(name,sucesscb,errorcb){
		
		postService.save({action:'getPostByName'},name,sucesscb,errorcb);
		
	};
	
	return 	postService;
}]);

File: src/com/cqut/service/post/PostService.java
package com.cqut.service.post;

import java.util.List;

import org.apache.ibatis.annotations.Param;

import com.cqut.entity.Post;

public interface PostService {
	
	public int insertPost(Post post);
	
	public int updatePost(Post post);
	
	public List<Post> getAllPosts(String where);
	
	public Post getPostById(int EId);
	
	public int deletePost(int EId);
	
	public List<Post> getPostByDepartmentIdAndName(int departmentId,String name);

	public List<Post> getPostByDepartmentId(int departmentId);
	
	public List<Post> getPostByName(String name);
}
